data_clean.Rmd

```diff
@@ -0,0 +1,255 @@
+```
+
+Rain and Snow measurements come in a categorical form - let's split out the categories into individual boolean columns.  Read [this documentation](http://www.audubon.org/sites/default/files/documents/cbc_report_field_definitions_2013.pdf) for more.
+```{r}
+data <- data %>%
+  mutate(
+    # Decode am_snow
+    snow_heavy_am = grepl("1", am_snow),
+    snow_light_am =  grepl("2", am_snow),
+    snow_none_am =  grepl("3", am_snow),
+    snow_unknown_am =  grepl("4", am_snow),
+    # Decode pm_snow
+    snow_heavy_pm = grepl("1", pm_snow),
+    snow_light_pm =  grepl("2", pm_snow),
+    snow_none_pm =  grepl("3", pm_snow),
+    snow_unknown_pm =  grepl("4", pm_snow),
+    # Create new snow variable - did day have snow level at any point?
+    snow_heavy_any = snow_heavy_am | snow_heavy_pm,
+    snow_light_any = snow_light_am | snow_light_pm,
+    snow_none_any = snow_none_am | snow_none_pm,
+    snow_unknown_any = snow_unknown_am | snow_unknown_pm,
+    
+    # Decode am_rain
+    rain_heavy_am = grepl("1", am_rain),
+    rain_light_am =  grepl("2", am_rain),
+    rain_none_am =  grepl("3", am_rain),
+    rain_unknown_am =  grepl("4", am_rain),
+    # Decode pm_rain
+    rain_heavy_pm = grepl("1", pm_rain),
+    rain_light_pm =  grepl("2", pm_rain),
+    rain_none_pm =  grepl("3", pm_rain),
+    rain_unknown_pm =  grepl("4", pm_rain),
+    # Create new rain variable - did day have rain level at any point?
+    rain_heavy_any = rain_heavy_am | rain_heavy_pm,
+    rain_light_any = rain_light_am | rain_light_pm,
+    rain_none_any = rain_none_am | rain_none_pm,
+    rain_unknown_any = rain_unknown_am | rain_unknown_pm
+  ) %>%
+  select(-am_snow, -pm_snow, -am_rain, -pm_rain)  # Remove the original, coded columns
+```
+# Subjective Data Cleanup
+The following code chunks involve more subjective transformations and edits than the above code - these are areas where folks are likely to have stronger opinions on what should be done.
+
+## Distances
+Let's convert all distance measurements to kilometers.  Note that distance values with a corresponding `distance_unit` of `NA` are not altered.
+```{r}
+data <- data %>%
+  mutate(
+    # Convert field_distance to kilometers
+    field_distance = if_else(
+      distance_units == "Miles",
+      1.60934 * field_distance,
+      field_distance
+      ),
+    # Convert nocturnal distance to kilometers
+    nocturnal_distance = if_else(
+      distance_units == "Miles",
+      1.60934 * nocturnal_distance,
+      nocturnal_distance
+      ),
+    # Now that all distances are in kilometeres, update `distance_units`, preserving NA values
+    distance_units = if_else(
+      distance_units == "Miles", 
+      "Kilometers",
+      distance_units
+      )
+    )
+```
+
+## Temperature
+Let's look at temperature. I can not definitively say whether the `temp_unit` column actually means anything, so for now we assume all temperatures are in Fahrenheit.
```

Comment:
The Temp Unit are: 
key = temp_unit
1 = celsius
2 = farenheit

Also: wind_unit key
1 = mph
3 = kmh

It looks like both need to be converted.